### PR TITLE
CDAP-1843 fix NaN issue on start time after upgrade

### PR DESCRIPTION
**Pull request by rohitsinha54**
The start time is just a part of the key in runRecordCompleted so get it from there 

Issue: https://issues.cask.co/browse/CDAP-1843
Build: http://builds.cask.co/browse/CDAP-RBT172-1

### CODE REVIEW

File: cdap-master/src/main/java/co/cask/cdap/data/tools/MDSUpgrader.java [hunk 1]
```diff
@@ -235,15 +236,16 @@ private void handleRunRecordCompleted(final String appId, final String programId
...
       @Override
       public Void apply(AppMDS appMetaStore) throws Exception {
-        List<RunRecord> completedRunRecords = appMetaStore.mds.list(partialKey, RunRecord.class);
-        for (RunRecord curCompletedRunRecord : completedRunRecords) {
+        Map<MDSKey, RunRecord> completedRunRecords = appMetaStore.mds.listKV(partialKey, RunRecord.class);
+        for (Map.Entry<MDSKey, RunRecord> curCompletedRunRecord : completedRunRecords.entrySet()) {
           // we cannot add a runRecordCompleted if there is no runRecordStarted for the run so write it first
           // the next setStop call deletes the runRecordStarted for that run so we don't have to delete it here.
-          writeTempRunRecordStart(appId, programType, programId, curCompletedRunRecord.getPid(),
-                                  curCompletedRunRecord.getStartTs());
```
Comment (anwar6953) [inline cdap-master/src/main/java/co/cask/cdap/data/tools/MDSUpgrader.java]: what did .getStartTs() return?

Comment (rohitsinha54) [inline cdap-master/src/main/java/co/cask/cdap/data/tools/MDSUpgrader.java]: We don't put anything when we write stop record. The function is there to get start time for RunRecord of started.

File: cdap-master/src/main/java/co/cask/cdap/data/tools/MDSUpgrader.java [hunk 2]
```diff
@@ -263,7 +263,7 @@ public Void apply(AppMDS appMetaStore) throws Exception {
   private void writeTempRunRecordStart(String appId, ProgramType programType, String programId, String pId,
                                        long startTs) {
     store.setStart(Id.Program.from(Id.Application.from(Constants.DEFAULT_NAMESPACE, appId), programType, programId),
-                   pId, Long.MAX_VALUE - startTs);
```
Comment (rohitsinha54) [inline cdap-master/src/main/java/co/cask/cdap/data/tools/MDSUpgrader.java]: Do not invert  the start time as the one store in completed run record is not inverted in the first place

**Comment by bdmogal** [comment]
:+1:

**Pull request merged**